datasetinsights/stats/coco_stats.py

```diff
@@ -0,0 +1,475 @@
+    for idx in coco_img_ids:
+        img = load_img_ann_for_single_image(coco_obj=coco_obj, img_id=idx)
+        max_height = max(img["height"], max_height)
+        max_width = max(img["width"], max_width)
+
+    bbox_heatmap = np.zeros([max_height, max_width, 1])
+    return bbox_heatmap
+
+
+def _get_labeled_kpt_dict(coco_obj: COCO, cat_id: int = 1) -> Dict:
+    """
+
+    Args:
+        coco_obj (pycocotools.coco.COCO): COCO object
+        cat_id (int): Category id, Default: 1 for person category
+
+    Returns:
+        Dict: Keypoint dictionary with initial values for each kp as 0.0
+
+    """
+    kpt_dict = {}
+    coco_keypoints = get_coco_keypoints(coco_obj=coco_obj, cat_id=cat_id)
+    for keypoint in coco_keypoints:
+        kpt_dict[keypoint] = 0.0
+    return kpt_dict
+
+
+def _get_bbox_relative_size(
+    bbox_area: float, image_h: float, image_w: float
+) -> float:
+    """
+
+    Args:
+        bbox_area (float): Bounding box area size
+        image_h (float): Image height
+        image_w (float): Image width
+
+    Returns:
+        float: Relative size of bbox w.r.t image size
+
+    """
+    bbox_relative_size = math.sqrt(bbox_area / (image_h * image_w))
+    return bbox_relative_size
+
+
+def get_labeled_keypoints_dict(
+    annotation_file: str = None, coco_obj: COCO = None
+) -> Dict:
+    """
+    Args:
+        annotation_file (JSON): COCO annotations json file path
+        coco_obj (pycocotools.coco.COCO): COCO object
+
+    Returns:
+        Dict: Labeled keypoints dictionary where key is the keypoint and and
+        val is the probability of that keypoint to occur in the bbox given
+        that kp is labeled.
+    """
+    if coco_obj:
+        coco = coco_obj
+    elif annotation_file:
+        coco = load_coco_annotations(annotation_file=annotation_file)
+    else:
+        raise ValueError(
+            "Must provide either annotation file or "
+            "pycocotools.coco.COCO object"
+        )
+    labeled_kpt_dict = _get_labeled_kpt_dict(coco_obj=coco)
+    keypoints = list(labeled_kpt_dict.keys())
+    img_ids = coco.getImgIds(catIds=1)
+    total_instances = 0
+
+    for idx in tqdm(img_ids):
+        img_ann = load_img_ann_for_single_image(coco_obj=coco, img_id=idx)
+        annotation_ids = coco.getAnnIds(
+            imgIds=img_ann["id"], catIds=1, iscrowd=None
+        )
+        annotations = coco.loadAnns(annotation_ids)
+        num_annotations = len(annotations)
+        total_instances += num_annotations
+
+        for person in range(num_annotations):
+
+            kp_visibility_flags = annotations[person]["keypoints"][2::3]
+            for i in range(len(keypoints)):
+                if kp_visibility_flags[i] != 0:
+                    labeled_kpt_dict[keypoints[i]] += 1
+
+    for key in labeled_kpt_dict.keys():
+        labeled_kpt_dict[key] = labeled_kpt_dict[key] / total_instances
+
+    return labeled_kpt_dict
+
+
+def get_bbox_heatmap(
```

Comment:
Let's prevent passing an instance of `coco_obj` around all methods. I recommend using primative python data structures (list, dictionary, nested dictionary,...) or python `datacalss` to represent data loaded. This will allow you to extend dataset statistics beyond COCO format, without re-implement every single method for dataset in different format (open-close principle)